### PR TITLE
Add launchDurationMillis config option

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Replaces the `launchCrashThresholdMs` configuration option with `launchDurationMillis`. This is consistent with the property name that will be used on bugsnag-cocoa.

## Changeset

- Added accessors for `launchDurationMillis` on `Configuration` and deprecated accessors for `launchCrashThresholdMs`, forwarding calls onto the new option
- Renamed internal properties to `ConfigInternal#launchDurationMillis` and `ImmutableConfig#launchDurationMillis`
- Updated manifest config loading to handle `com.bugsnag.android.LAUNCH_DURATION_MILLIS` values, while maintaining backwards compatibility with the old `launchCrashThresholdMs` value

## Testing

Added equivalent unit test coverage for `Configuration` implementation and loading from the manifest as existed for `launchCrashThresholdMs`.

### CODE REVIEW

File: bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ConfigSerializer.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public void serialize(Map<String, Object> map, ImmutableConfig config) {
         map.put("versionCode", config.getVersionCode());
         map.put("type", config.getAppType());
         map.put("persistUser", config.getPersistUser());
-        map.put("launchCrashThresholdMs", (int) config.getLaunchCrashThresholdMs());
+        map.put("launchCrashThresholdMs", (int) config.getLaunchDurationMillis());
```
Comment (tomlongridge) [inline bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ConfigSerializer.java:28]: Does this require a React Native update to pass the renamed config field?

File: bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public int compare(File lhs, File rhs) {
     }
 
     void flushOnLaunch() {
-        if (config.getLaunchCrashThresholdMs() != 0) {
+        if (config.getLaunchDurationMillis() != 0) {
```
Comment (tomlongridge) [inline bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java:64]: I assume this is just a like-for-like change and we're going to amend this behaviour in a future change on the project wrt to sending launch crashes synchronously being controlled by the new config option?

File: bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ConfigSerializer.java (hunk 1 above)
Comment (tomlongridge) [reply bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/ConfigSerializer.java:28]: Discussed in slack.

File: bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java (hunk 2 above)
Comment (fractalwrench) [reply bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java:64]: Yes - this behaviour will be altered in future changesets.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## TBD
+
+### Enhancements
+
+* Add launchDurationMillis config option
```
[the remark this answers is not in the document]
Comment (fractalwrench) [reply CHANGELOG.md]: Tweaked changelog to add deprecations section.

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**